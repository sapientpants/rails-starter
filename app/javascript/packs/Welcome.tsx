import * as React from 'react';
import * as ReactDOM from 'react-dom';

interface WelcomeProps {
  message: string;
}

const Welcome: React.FC<WelcomeProps> = ({ message }) => {
  return (
    <button className="btn btn-primary" onClick={() => alert(message)}>
      Click on me!
    </button>
  )
}

document.addEventListener('DOMContentLoaded', () => {
  ReactDOM.render(
    <Welcome message="Hello, kind user!" />,
    document.getElementById('welcome-button')
  )
})