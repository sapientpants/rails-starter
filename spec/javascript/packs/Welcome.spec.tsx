import React from 'react';
import renderer from 'react-test-renderer';
import Welcome from '../../../app/javascript/packs/Welcome';

it('renders correctly', () => {
  const tree = renderer
    .create(<Welcome message="Hello World!" />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});