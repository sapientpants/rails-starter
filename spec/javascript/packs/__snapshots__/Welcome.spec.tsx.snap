// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<button
  className="btn btn-primary"
  onClick={[Function]}
>
  Click on me!
</button>
`;
